### ISSUE TITLE
Getting an error when deleting an object immediately after creating it.

### ISSUE THREAD

**Issue by Siddhant085** [societyserver/sTeam#109]
usr/local/lib/steam/tmp/net/coal/protocoll.pike:174: /kernel/securesocket()->COAL_command(42,Server Object,({}))
usr/local/lib/steam/tmp/net/coal.pike:175: /kernel/securesocket()->receive_message("")
usr/local/lib/steam/tmp/kernel/securesocket.pike:30: /kernel/securesocket()->receive_message("�\0\0\0#\0\0\0_\1\0\0\0\0\0\0\0\0\5\0\2\3\0\0\0\6delete\5\0\0")
usr/local/lib/steam/tmp/kernel/coalsocket.pike:68: /kernel/securesocket()->f_read_callback(0,"�\0\0\0#\0\0\0_\1\0\0\0\0\0\0\0\0\5\0\2\3\0\0\0\6delete\5\0\0")
-:1: Pike.Backend(0)->`()(3600.0)
usr/local/pike/7.8.866/lib/master.pike:3908: master()->_main(({"steam","server/server.pike","--pid=/var/run/steam.pid","--quiet","--auto-restart"}))

delete:Bad line array
1
Function: delete not found inside (0)
192

This error is occurring in the create test case sometimes but not always. The object is being deleted immediately after creation so this error shouldn't be there.

**Comment by eMBee** [comment]
what is the code that leads to this error?

if you replace delete with some other function from the object. do you
get the same?

the error effectively claims that a call is made on 0->delete()
which means, the object reference is broken, and returns 0, and
therefore any access to the object should fail.

try the following:
wait a few seconds. does the problem still occur?

restart the client (Reconnect). does that work?

if the object is permanently broken, does restarting the server fix it?

**Comment by Siddhant085** [comment]
I am getting the error for other call also. I tried query_attribute instead of delete. The problem persists.

The code is:
int result =_Server->get_factory("Calendar")->execute((["name":"TestCalendar"]))->move(OBJ(me->get_last_trail()->query_attribute("OBJ_PATH"))); 
OBJ(me->get_last_trail()->query_attribute("OBJ_PATH")+"/TestCalendar")->query_attribute("OBJ_NAME");

The error does not occur 100% of the times I run the code. Sometimes restarting the server or the client fixes the error but it comes back on repeated execution.

**Comment by eMBee** [comment]
maybe there is a problem with the trail.
try instead saving a reference to the object, and access it directly, or
access it by path.

trail should only be used to test the trail function.
also, trail only holds rooms. no other objects.

**Comment by Siddhant085** [comment]
object room = OBJ("/TestRoom");
int result =_Server->get_factory("Calendar")->execute((["name":"TestCalendar"]))->move(room); 
OBJ("TestRoom/TestCalendar")->delete();

I changed the code to this and removed all references to get _trail, however the error persists.

**Comment by eMBee** [comment]
can you reploduce the problem with any other class? if not, then that
suggests that the Calendar class has an issue. it is not actively used,
so if that's the case i'd leave it as a failing test.

**Comment by Siddhant085** [comment]
No the problem is not only for the calendar class. It is occurring quite randomly. I am creating 4 objects Document, Calendar, Date and Container and the error occurs for any one of the class and that also not always so reproducing the problem is difficult.

**Comment by Siddhant085** [comment]
@eMBee This problem is occurring not only with the delete function but also other functions like get_class(). In fact I changed the code so as to make sure that the object exists.
`_Server->get_factory("Room")->execute((["name":"move2Room"]))->move(OBJ("/TestRoom"));
    _Server->get_factory("User")->execute((["name":"move2User","pw":"testpass","email":"abc@example.com"]));
    _Server->get_module("users")->get_user("move2User")->activate_user();
    array(object) testObjects = allocate(2);
    do{
    testObjects[0]=OBJ("/TestRoom/move2Room");
    }while(testObjects[0]==0);
    do{
    testObjects[1]=_Server->get_module("users")->get_user("move2User");
    }while(testObjects[1]==0);
`
Even after this when I call testObjects[0]->get_class(), I am getting the error Function: get_class not found inside (0).

**Comment by eMBee** [comment]
this is actually what i expected. somehow the reference to the object
disappears.

does it come back if you restart the client?
if not, does it come back if you restart the server?

**Comment by Siddhant085** [comment]
Yes this behavior is quite random sometimes it comes back on restarting the client and sometimes on restarting the server.
